src/creature_instances.c

```diff
@@ -537,6 +548,133 @@ long process_creature_self_spell_casting(struct Thing* creatng)
     return 1;
 }
 
+/**
+ * @brief Check whether the given creature is suitable to cast ranged heal spell.
+ * This function is used for both combat and non-combat situations.
+ *
+ * @param creatng The creature being checked.
+ * @return CrInstance The instance index being set.
+ */
+CrInstance process_creature_ranged_heal_spell_casting(struct Thing* creatng)
```

Comment:
I would prefer this to be `process_creature_ranged_buff_spell_casting` or something similar. Then have 'Heal' as a subfunction with the conditions of 'need healing'.

Reply:
Done.

Reply:
Currently `process_creature_ranged_buff_spell_casting` is still focused purely on 'find a target to heal'. I meant more that it could be turned into a general 'do ranged buff' function, and have the 'best choice' part, be a sub-function specific to that instance.

Reply:
I can do that if you want a more generic design in this PR.